lib/ramble/ramble/main.py

```diff
@@ -116,29 +116,42 @@ def get_version():
 
     The commit sha is only added when available.
     """
-    import spack.util.git
     version = ramble.ramble_version
-    git_path = os.path.join(ramble.paths.prefix, ".git")
+    git_hash = get_git_hash(path=ramble.paths.prefix)
+
+    if git_hash:
+        version += f' ({git_hash})'
+
+    return version
+
+
+def get_git_hash(path=ramble.paths.prefix):
+    """Get get hash from a path
+
+    Outputs '<git commit sha>'.
+    """
+    import spack.util.git
+    git_path = os.path.join(path, ".git")
     if os.path.exists(git_path):
         git = spack.util.git.git()
         if not git:
-            return version
+            return None
         rev = git(
             "-C",
-            ramble.paths.prefix,
+            path,
             "rev-parse",
             "HEAD",
             output=str,
             error=os.devnull,
             fail_on_error=False,
         )
         if git.returncode != 0:
-            return version
+            return None
         match = re.match(r"[a-f\d]{7,}$", rev)
         if match:
-            version += " ({0})".format(match.group(0))
+            return match.group(0)
 
-    return version
+    return None
```

Comment:
Is the none value meaningful here? Should we just be returning or continuing?

Reply:
I changed it to just a return.